apps/studio/pages/api/ai/sql/tools.ts

```diff
@@ -387,5 +387,179 @@ export const getTools = ({
+          const { name }: reqPayload = await req.json();
+          const data = {
+            message: \`Hello \${name} from foo!\`,
+          };
+
+          return new Response(
+            JSON.stringify(data),
+            { headers: { 'Content-Type': 'application/json', 'Connection': 'keep-alive' }}
+          );
+        });
+        \`\`\`
+
+        ### Example Function using Node built-in API
+
+        \`\`\`edge
+        import { randomBytes } from "node:crypto";
+        import { createServer } from "node:http";
+        import process from "node:process";
+
+        const generateRandomString = (length) => {
+          const buffer = randomBytes(length);
+          return buffer.toString('hex');
+        };
+
+        const randomString = generateRandomString(10);
+        console.log(randomString);
+
+        const server = createServer((req, res) => {
+          const message = \`Hello\`;
+          res.end(message);
+        });
+
+        server.listen(9999);
+        \`\`\`
+
+        ### Using npm packages in Functions
+
+        \`\`\`edge
+        import express from "npm:express@4.18.2";
+
+        const app = express();
+
+        app.get(/(.*)/, (req, res) => {
+          res.send("Welcome to Supabase");
+        });
+
+        app.listen(8000);
+        \`\`\`
+
+        ### Generate embeddings using built-in @Supabase.ai API
+
+        \`\`\`edge
+        const model = new Supabase.ai.Session('gte-small');
+
+        Deno.serve(async (req: Request) => {
+          const params = new URL(req.url).searchParams;
+          const input = params.get('text');
+          const output = await model.run(input, { mean_pool: true, normalize: true });
+          return new Response(
+            JSON.stringify(output),
+            {
+              headers: {
+                'Content-Type': 'application/json',
+                'Connection': 'keep-alive',
+              },
+            },
+          );
+        });
+        \`\`\`
+
+        ## Integrating with Supabase Auth
+
+        \`\`\`edge
+          import { createClient } from \\'jsr:@supabase/supabase-js@2\\'
+          import { corsHeaders } from \\'../_shared/cors.ts\\'
+
+          console.log(\`Function "select-from-table-with-auth-rls" up and running!\`)
+
+          Deno.serve(async (req: Request) => {
+            // This is needed if you\\'re planning to invoke your function from a browser.
+            if (req.method === \\'OPTIONS\\') {
+              return new Response(\\'ok\\', { headers: corsHeaders })
+            }
+
+            try {
+              // Create a Supabase client with the Auth context of the logged in user.
+              const supabaseClient = createClient(
+                // Supabase API URL - env var exported by default.
+                Deno.env.get(\\'SUPABASE_URL\\') ?? \\'\\',
+                // Supabase API ANON KEY - env var exported by default.
+                Deno.env.get(\\'SUPABASE_ANON_KEY\\') ?? \\'\\',
```

Comment:
Why fall back to empty string? Better to surface missing vars as soon as possible right?

Reply:
@SaxonF is this an example you added? Maybe the original example had empty strings, but I agree better to let it leave as undefined so it throws an error in case the env vars are missing.